### PR TITLE
Reimplement "mackerel_aws_integration" data source with terraform-plugin-framework

### PR DESCRIPTION
**Pull request by tosuke**
Part of #243 . This PR adds a new "mackerel_aws_integration" data source implementation using `AWSIntegrationModel` introduced in #244 .

Output from acceptance testing:

```
$ MACKEREL_EXPERIMENTAL_TFFRAMEWORK=1 make testacc TESTS='"AccDataSourceMackerelAWSIntegration"'
TF_ACC=1 go test -v ./mackerel/... -run "AccDataSourceMackerelAWSIntegration" -timeout 120m
2024/10/31 20:40:44 [INFO] mackerel: use terraform-plugin-framework based implementation
=== RUN   TestAccDataSourceMackerelAWSIntegrationIAMRole
=== PAUSE TestAccDataSourceMackerelAWSIntegrationIAMRole
=== RUN   TestAccDataSourceMackerelAWSIntegrationCredential
=== PAUSE TestAccDataSourceMackerelAWSIntegrationCredential
=== CONT  TestAccDataSourceMackerelAWSIntegrationIAMRole
=== CONT  TestAccDataSourceMackerelAWSIntegrationCredential
--- PASS: TestAccDataSourceMackerelAWSIntegrationCredential (4.05s)
--- PASS: TestAccDataSourceMackerelAWSIntegrationIAMRole (4.34s)
PASS
ok      github.com/mackerelio-labs/terraform-provider-mackerel/mackerel 5.223s
```

### CODE REVIEW

File: internal/mackerel/aws_integration.go [hunk 1]
```diff
@@ -22,6 +22,25 @@ type AWSIntegrationModel struct {
...
 
+	AWSIntegrationSerfvices
+}
+
+type AWSIntegrationDataSourceModel struct {
+	ID           types.String `tfsdk:"id"`
+	Name         types.String `tfsdk:"name"`
+	Memo         types.String `tfsdk:"memo"`
+	Key          types.String `tfsdk:"key"`
+	SecretKey    types.String `tfsdk:"-"`
```
Comment (tosuke) [inline internal/mackerel/aws_integration.go:33]: AWS Integration resources contain an AWS secret key in their state, whereas data sources do not.

File: internal/provider/data_source_mackerel_aws_integration.go [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+		},
+		"included_tags": schema.StringAttribute{
+			Description: schemaAWSIntegrationIncludedTagsDesc,
+			Computed:    true,
+		},
+		"excluded_tags": schema.StringAttribute{
+			Description: schemaAWSIntegrationExcludedTagsDesc,
+			Computed:    true,
+		},
+	}
```
Comment (tosuke) [inline internal/provider/data_source_mackerel_aws_integration.go:101]: Original schema: https://github.com/mackerelio-labs/terraform-provider-mackerel/blob/0f8684217b72450e7a2aa3fdd3b6109f8244bce3/mackerel/data_source_mackerel_aws_integration.go?plain=1#L70-L108

File: internal/provider/data_source_mackerel_aws_integration.go [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+					AttrTypes: map[string]attr.Type{
+						"enable": types.BoolType,
+						"role":   types.StringType,
+						"excluded_metrics": types.ListType{
+							ElemType: types.StringType,
+						},
+					},
+				},
+			}
+		}
```
Comment (tosuke) [inline internal/provider/data_source_mackerel_aws_integration.go:131]: Original schema: https://github.com/mackerelio-labs/terraform-provider-mackerel/blob/0f8684217b72450e7a2aa3fdd3b6109f8244bce3/mackerel/data_source_mackerel_aws_integration.go?plain=1#L11-L59

**Review by yohfee — COMMENTED** [review]
It looks almost ok, but I have a tiny question.

File: internal/provider/data_source_mackerel_aws_integration.go [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+		},
+		"key": schema.StringAttribute{
+			Description: schemaAWSIntegrationKeyDesc,
+			Computed:    true,
+		},
+		"role_arn": schema.StringAttribute{
+			Description: schemaAWSIntegrationRoleARNDesc,
+			Computed:    true,
+		},
+		"external_id": schema.StringAttribute{
```
Comment (yohfee) [inline internal/provider/data_source_mackerel_aws_integration.go:85]: Is it ok to omit `Sensitive: true` ?

Comment (tosuke) [reply internal/provider/data_source_mackerel_aws_integration.go:85]: not ok... fixed it: 68f021b859990a888d29319d239669ae5653d10b

**Review by yohfee — APPROVED** [review]
Thanks, LGTM!

**Pull request merged**